# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../tse3/oss.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -urN tse3-0.3.1.orig/src/tse3/plt/OSS.h tse3-0.3.1/src/tse3/plt/OSS.h
--- tse3-0.3.1.orig/src/tse3/plt/OSS.h  2007-08-22 09:30:45.000000000 +0530
+++ tse3-0.3.1/src/tse3/plt/OSS.h       2007-08-22 09:31:02.000000000 +0530
@@ -253,4 +253,28 @@
     }
 }

+#define _AWE_MODE_FLAG                 0x80
+#define _AWE_SET_CMD(p,dev,voice,cmd,p1,p2) \
+{((char*)(p))[0] = SEQ_PRIVATE;\
+ ((char*)(p))[1] = dev;\
+ ((char*)(p))[2] = _AWE_MODE_FLAG|(cmd);\
+ ((char*)(p))[3] = voice;\
+ ((unsigned short*)(p))[2] = p1;\
+ ((unsigned short*)(p))[3] = p2;}
+
+#define _AWE_CMD(dev, voice, cmd, p1, p2) \
+{_SEQ_NEEDBUF(8);\
+ _AWE_SET_CMD(_seqbuf + _seqbufptr, dev, voice, cmd, p1, p2);\
+ _SEQ_ADVBUF(8);}
+#define _AWE_DRUM_CHANNELS             0x0b
+#define AWE_DRUM_CHANNELS(dev,channels) _AWE_CMD(dev, 0, _AWE_DRUM_CHANNELS, ((channels) & 0xffff), ((channels) >> 16))
+#define _AWE_CHANNEL_MODE              0x0a
+#define AWE_PLAY_MULTI         1       /* multi note voice mode */
+#define AWE_SET_CHANNEL_MODE(dev,mode) _AWE_CMD(dev, 0, _AWE_CHANNEL_MODE, mode, 0)
+#define _AWE_TERMINATE_ALL             0x07
+#define AWE_TERMINATE_ALL(dev) _AWE_CMD(dev, 0, _AWE_TERMINATE_ALL, 0, 0)
+#ifndef SAMPLE_TYPE_AWE32
+#define SAMPLE_TYPE_AWE32      0x20
+#endif
+
 #endif

